src/shared/settingsConfiguration.ts

```diff
@@ -4,38 +4,42 @@
  */
 
 import * as vscode from 'vscode'
-import { getLogger } from './logger'
 import * as packageJson from '../../package.json'
+import { ClassToInterfaceType } from './utilities/tsUtils'
+import { ext } from './extensionGlobals'
+import { isCI } from './vscode/env'
+import * as logger from './logger'
 
 /**
  * Wraps the VSCode configuration API and provides Toolkit-related
  * configuration functions.
  */
-export interface SettingsConfiguration {
-    readSetting<T>(settingKey: string): T | undefined
-    readSetting<T>(settingKey: string, defaultValue: T): T
+export type SettingsConfiguration = ClassToInterfaceType<DefaultSettingsConfiguration>
 
-    // array values are serialized as a comma-delimited string
-    /**
-     * Sets a config value.
-     *
-     * Writing to the (VSCode) config store may fail if the user does not have
-     * write permissions, or if some requirement is not met.  For example, the
-     * `vscode.ConfigurationTarget.Workspace` scope requires a workspace.
-     *
-     * @param settingKey  Config key name
-     * @param value  Config value
-     * @param target  Config _scope_
-     *
-     * @returns true on success, else false
-     */
-    writeSetting<T>(settingKey: string, value: T | undefined, target: vscode.ConfigurationTarget): Promise<boolean>
-}
+export type AwsDevSetting = 'aws.forceCloud9' | 'aws.developer.foo1' | 'aws.developer.foo2'
+
+type jsPrimitiveTypeName =
```

Comment:
```suggestion
type JSPrimitiveTypeName =
```
Types/interfaces are always capitalized by convention